guides/using-references.mdx

```diff
@@ -0,0 +1,237 @@
+    enabled: false
+```
+
+<Note>
+  You don't have to call your file `features.yml` and you can spread your flag
+  definitions across multiple files. Checkout our docs on [locating flag
+  state](/configuration/storage#locating-flag-state) to learn more.
+</Note>
+
+## Creating Preview Environments
+
+For this guide, we're going to use [GitHub Actions](https://github.com/features/actions) to deploy our preview environments to [Koyeb](https://www.koyeb.com/).
+
+Koyeb has a nice tutorial on how to deploy a preview environment using GitHub Actions. You can find it [here](https://www.koyeb.com/tutorials/deploy-preview-environments-on-koyeb-for-github-pull-requests).
+
+To get our preview environments working with Flipt, we'll need some way of passing the Git branch name to Flipt. We can do this by setting the `FLIPT_CLIENT_REFERENCE` environment variable to the Git branch name. This will allow us to serve a different Git branch for each preview environment.
+
+To do this, we'll need to add a step to our GitHub Actions workflow that sets the `FLIPT_CLIENT_REFERENCE` environment variable to the Git branch name.
+
+```yaml
+- name: Deploy the application to Koyeb
+  uses: koyeb/action-git-deploy@v1
+  with:
+    app-name: "dashboard-app-preview-${{ github.head_ref }}"
+    service-name: ${{ github.head_ref }}
+    service-ports: "8081:http"
+    service-routes: "/:8081"
+    service-env: "FLIPT_CLIENT_REFERENCE=${{ github.head_ref }},FLIPT_ADDRESS=${{ secrets.FLIPT_ADDRESS }}"
+    docker: ghcr.io/flipt-io/dashboard-app:latest
+```
+
+Here we're also setting the `FLIPT_ADDRESS` environment variable to the address of our Flipt instance. This could also be configured in your application via a config file.
+
+## Enabling the Flag and Pushing to a Preview Branch
+
+Now that we have our preview environments setup, we can enable our feature flag and push our changes to a preview branch.
+
+To enable our feature flag, we'll need to update our `features.yml` file to set the `enabled` field to `true`.
+
+```yaml pkg/performance/features.yml
+namespace: performance
+flags:
+  - key: showPerformanceHistory
+    name: Show Performance History Graph
+    type: BOOLEAN_FLAG_TYPE
+    enabled: true
+```
+
+Now we can add, commit, and push our changes to a preview branch.
+
+```bash
+git checkout -b feature/enable-performance-history
+git add pkg/performance/features.yml
+git commit -m "Enable performance history"
+git push origin feature/enable-performance-history
+```
+
+We'll also need to create a pull request for our preview branch. This will trigger our GitHub Actions workflow and deploy our preview environment.
+
+## Testing the Change
+
+Before checking on our preview environment, let's take a look at our Flipt instance. We can see that our feature flag is still showing as disabled in the UI.
+
+This is because Flipt is configured to track the `main` branch of our Git repository by default which is what it continues to show in the UI.
```

Comment:
```suggestion
This is because Flipt is configured to track the `main` branch of our Git repository by default, which is what it continues to show in the UI.
```